core/src/test/scala/akka/kafka/scaladsl/IntegrationSpec.scala

```diff
@@ -197,9 +208,70 @@ class IntegrationSpec extends TestKit(ActorSystem("IntegrationSpec"))
       Await.result(control.isShutdown, remainingOrDefault)
     }
 
-    // TODO add more tests
+    "consume and commit in batches" in {
+      givenInitializedTopic()
 
-  }
+      produceMessages(topic1, 1 to 100)
+
+      val consumerSettings = createConsumerSettings(topic1, group1, client1)
 
+      def consumeAndBatchCommit() = {
+        Consumer.committableSource(consumerSettings)
+          .map { msg => { ; msg.committableOffset } }
+          .batch(max = 10, first => CommittableOffsetBatch.empty.updated(first)) { (batch, elem) =>
+            batch.updated(elem)
+          }
+          .mapAsync(1)(_.commitScaladsl()).toMat(TestSink.probe)(Keep.both)
+      }
+
+      consumeAndBatchCommit()
+
+      val probe = createProbe(consumerSettings)
+
+      probe
+        .request(100)
+        .expectNextN((1 to 100).map(_.toString))
+
+      produceMessages(topic1, 101 to 150)
+
+      consumeAndBatchCommit()
```

Comment:
at this point we have two concurrent connections to same topic. should we stop previous one?